aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -100,7 +100,7 @@ static void copy_cast_mps(at::Tensor& dst,
     MTLResourceOptions options = MTLResourceCPUCacheModeDefaultCache | MTLResourceStorageModeShared;
     NSUInteger alignedLength = 0;
 
-    const void* host_dst = dst.storage().data();
+    const void* host_dst = static_cast<const char*>(dst.storage().data()) + dst.storage_offset() * dst.itemsize();
```

Comment:
This is a pretty invasive change 😄 . I think it makes sense. 
I will let @DenisVieriu97 comment on this...

Reply:
We use the same trick here: https://github.com/pytorch/pytorch/blob/11a9548cbb7661b0a20628ef5227a9680e7bf14b/aten/src/ATen/native/mps/operations/Copy.mm#L153-L163

I could probably do it more elegantly with `getMTLBufferStorage()` so it would not look so invasive :). But would like to let @DenisVieriu97 comment first.